fs/reader/reader.go

```diff
@@ -227,14 +227,15 @@ func (sf *file) ReadAt(p []byte, offset int64) (int, error) {
 		return 0, errors.Wrap(err, "failed to read the file")
 	}
 
-	commonmetrics.IncOperationCount(commonmetrics.OnDemandRemoteRegistryFetchCount, sf.gr.layerSha) // increment the number of on demand file fetches from remote registry
+	// TODO this is not the right place for this metric to be. It needs to go down the BlobReader, when the HTTP request is issued
```

Comment:
HTTP requests can be issued from the bg-fetcher as well. How will the blobReader differentiate between the two?

Reply:
I'm just stating that this is not the right place for this metric to be. #241 speaks more about revisiting this notion and to think if this is really needed.